src/main/java/com/cosmian/jna/findex/Findex.java

```diff
@@ -65,111 +66,62 @@ public static void set_error(String error_msg) throws CloudproofException {
+    private static String indexedValuesToJson(Map<IndexedValue, Set<Keyword>> indexedValuesAndWords)
         throws CloudproofException {
-
         // For the JSON strings
         ObjectMapper mapper = new ObjectMapper();
+        Encoder encoder = Base64.getEncoder();
+        HashMap<String, String[]> indexedValuesAndWordsString = new HashMap<>();
+        for (Entry<IndexedValue, Set<Keyword>> entry : indexedValuesAndWords.entrySet()) {
+            String[] words = new String[entry.getValue().size()];
+            int i = 0;
+            for (Keyword word : entry.getValue()) {
+                words[i++] = encoder.encodeToString(word.getBytes());
+            }
+            indexedValuesAndWordsString.put(
+                encoder.encodeToString(entry.getKey().getBytes()),
+                words);
+        }
 
-        // Findex master keys
-        String masterKeysJson;
+        String indexedValuesAndWordsJson;
         try {
-            masterKeysJson = mapper.writeValueAsString(masterKeys);
+            indexedValuesAndWordsJson = mapper.writeValueAsString(indexedValuesAndWordsString);
         } catch (JsonProcessingException e) {
-            throw new CloudproofException("Invalid master keys", e);
-        }
-
-        try (final Memory labelPointer = new Memory(label.length)) {
-            labelPointer.write(0, label, 0, label.length);
-
-            // Findex indexed values and words
-            HashMap<String, String[]> indexedValuesAndWordsString = new HashMap<>();
-            for (Entry<IndexedValue, Word[]> entry : indexedValuesAndWords.entrySet()) {
-                String[] words = new String[entry.getValue().length];
-                int i = 0;
-                for (Word word : entry.getValue()) {
-                    words[i++] = word.toString();
-                }
-                indexedValuesAndWordsString.put(entry.getKey().toString(), words);
-            }
-
-            String indexedValuesAndWordsJson;
-            try {
-                indexedValuesAndWordsJson = mapper.writeValueAsString(indexedValuesAndWordsString);
-            } catch (JsonProcessingException e) {
-                throw new CloudproofException("Invalid indexed values and words", e);
-            }
-
-            // Indexes creation + insertion/update
-            unwrap(Findex.INSTANCE.h_graph_upsert(masterKeysJson, labelPointer, label.length, indexedValuesAndWordsJson,
-                fetchEntry,
-                upsertEntry, upsertChain));
+            throw new CloudproofException("Invalid indexed values and words", e);
         }
+        return indexedValuesAndWordsJson;
     }
 
-    public static List<byte[]> search(byte[] keyK, byte[] label, Word[] words, int loopIterationLimit, int maxDepth,
-        ProgressCallback progress, FetchEntryCallback fetchEntry, FetchChainCallback fetchChain)
+    public static Map<Keyword, Set<Location>> search(byte[] key,
```

Comment:
You use `keyWords` and `keyword`, maybe standardize the casing with `keywords`

Reply:
This must be fixed now: I could not find any occurrence of KeyWord